nav-challenge/src/components/pages/signUpPage/SignUpPage.tsx

```diff
@@ -0,0 +1,27 @@
+import { FC, useContext } from "react";
+import { SignUpFormProps } from "../../../utils/types/SignUpFormProps";
+import { SignUpContext } from "../../containers/signUpContainer/SignUpContainer";
+import SignInSignUpLayout from "../../layout/signUpSignInLayout/SignInSignUpLayout";
+import SignUpForm from "../../signUpForm/SignUpForm";
+import "./SignUpPage.scss";
+
+const SignUpPage: FC = () => {
+  const signUpData: SignUpFormProps = useContext(SignUpContext);
```

Comment:
Couldn't have this been passed as a prop instead of via context?

Reply:
this is the question i was going to ask you. This SignUpPage is rendering SignUpForm and that form is using all the props. So my question/idea is  to use that context in child component not in parent.


could i add that context in child component (SignUpForm) and remove useContext from SIgnUpPage?

i hope i was clear enough :) if not we can discuss on slack a bit more..

Reply:
It's ok if we want to use the context instead of passing props and in that case the context should be used in the component which is the first common parent for all clients of the provider.
Lets say we have components `Button`, `Input` and `Title` that are all using the same context, and we have the following structure:
```
<GrandParent>
  <Parent>
    <Title />
    <Button />
    <Input />
  </Parent>
</GrandParent>
```
then the context should be in the `Parent` component.
However if we have a structure like:
```
<GrandParent>
  <Title />
  <Parent>
    <Button />
    <Input />
  </Parent>
</GrandParent>
```
then the context is in the `GrandParent` component.
I hope that my answer has been clear enough :)